packages/svelte/src/lib/components/primitives/AmplifyFormField.svelte

```diff
@@ -0,0 +1,145 @@
+<script lang="ts">
+	import AmplifyPhoneNumberField from './AmplifyPhoneNumberField.svelte';
+	import AmplifyPasswordField from './AmplifyPasswordField.svelte';
+	import AmplifyTextField from './AmplifyTextField.svelte';
+	import AmplifyError from './AmplifyError.svelte';
+
+	import { authState, updateForm, updateBlur } from '$lib/components/store';
+	import {
+		ActorContextWithForms,
+		authInputAttributes,
+		AuthInputAttributes,
+		getActorContext,
+		translate
+	} from '@aws-amplify/ui';
+
+	import { onMount } from 'svelte';
+
+	export let name: string;
+	export let type: string;
+	export let required = true;
+	export let placeholder = '';
+	export let label = '';
+	export let initialValue = '';
+	export let disabled = false;
+	export let autocomplete = '';
+	export let labelHidden = true;
+
+	const getAttributeMap: AttributeInfoProvider = () => authInputAttributes;
+	type AttributeInfoProvider = () => AuthInputAttributes;
+
+	let defaultCountryCode: string;
+	onMount(() => {
+		if (isPhoneField) {
+			const state = $authState;
+			const { country_code }: ActorContextWithForms = getActorContext(state);
+			defaultCountryCode = country_code;
+
+			// TODO: remove this side-effect
+			updateForm({
+				name: 'country_code',
+				value: country_code
+			});
+		}
+	});
+
+	const attributeMap = getAttributeMap();
+
+	let error: string;
+	$: {
+		const formContext: ActorContextWithForms = getActorContext($authState);
+		const { validationError } = formContext;
+		error = translate(validationError[name]);
+	}
+
+	function onBlur($event) {
+		let { name } = <HTMLInputElement>$event?.detail?.target;
+
+		updateBlur({ name });
+	}
+
+	let inferLabel: string;
+	$: {
+		const myLabel = label || attributeMap[name]?.label;
+		inferLabel = translate<string>(myLabel);
+	}
```

Comment:
Minor nit/non-blocking, does this need to be reactive? I imagine `label` to be static once it's mounted.

Reply:
I wanted to make sure it updates if the translation changes.

Reply:
Ah, you're right.